pyracing/client.py

```diff
@@ -287,10 +321,18 @@ async def driver_stats(
         }
         url = ct.URL_DRIVER_STATS
         response = await self._build_request(url, payload)
-
-        return [historical_data.DriverStats(x) for
-                x in response.json()["d"]["r"]]
-
+        
+        if not response.json():
+            return []
+        
+        mapping = response.json().get('m')
+        response_items = response.json().get('d').get('r')
+        
+        if mapping and response_items:
+            renamed_items = [self._rename_numerical_keys(ri, mapping) for ri in response_items]
+            
+            return [historical_data.DriverStats(x) for x in renamed_items]
```

Comment:
If we don't have a mapping and response_items here then this just won't return, we should probably return an empty list at least

Reply:
Updated the methods that are calling _rename_numerical_keys to return an empty list.